### PR TITLE
unarchive: fix work with 0 bytes archives

### LINKED ISSUE #1575: unarchive: don't try to work on 0 byte files
When you call unarchive on an archive that's 0 bytes it will report "Failed to find handler to unarchive. Make sure the required command to extract the file is installed.", which is rather misleading.

A solution forward might be to check for 0 byte size before trying to work on the file.

### PR DESCRIPTION
**Pull request by verm666**
This change is to fix #1575

### CODE REVIEW

**Comment by gregdek** [comment]
@pileofrogs please review; if you give +1 we will promote to P1_reviewed for faster inclusion.

File: files/unarchive.py [hunk 1]
```diff
@@ -300,6 +300,16 @@ def main():
     if not os.access(src, os.R_OK):
         module.fail_json(msg="Source '%s' not readable" % src)
 
+    # skip working with 0 size archives
+    try:
+        if os.path.getsize(src) == 0:
+            res_args = {
+                'changed': False
```
Comment (bcoca) [inline files/unarchive.py]: ~~I would add msg="0 byte archive found, nothing to do", so users are not puzzled by the lack of change.~~

actually this should be an error: invalid archive, the file is 0 bytes.

File: files/unarchive.py [hunk 2]
```diff
@@ -300,6 +300,16 @@ def main():
...
+            }
+            module.exit_json(**res_args)
+    except Exception, e:
+        module.fail_json(msg="Source '%s' not readable" % src)
```
Comment (bcoca) [inline files/unarchive.py:308]: append exception message to failure message

**Comment by ashemedai** [comment]
I am not fully versed in current Ansible coding practices, but this looks like a very clean fix for the problem. Thanks! Hadn't had time yet so far to work on it myself, much appreciated.

**Comment by gregdek** [comment]
@pileofrogs can you review?

**Pull request merged**